src/include/platform/internal/GenericPlatformManagerImpl.cpp

```diff
@@ -66,7 +66,7 @@ CHIP_ERROR GenericPlatformManagerImpl<ImplClass>::_InitChipStack()
     SuccessOrExit(err);
 
     // Initialize the CHIP system layer.
-    new (&SystemLayer) System::Layer();
+    new (&SystemLayer) System::LayerImpl();
```

Comment:
why do we need placement new here?

should we have some guarantee that sizes will match? what ensures that sizeof(LayerImpl) >= sizeof(SystemLayer) ?

Reply:
This confuses me .... what is SystemLayer? Searching through the code and cannot sem to find it and I do not see either a g or m prefix to figure out if a global or member or something else.

Reply:
Yeah, good question; I missed this `new` because I didn't actually touch this beyond name replacement.

`SystemLayer` is defined in `src/platform/Globals.cpp`. It's a `LayerImpl` so the placement `new` is safe, but seems redundant (subject to this being ~Monday~ Tuesday morning) since the global constructor would have run.

I have a followup in which I replace all the many uses of the `DeviceLayer::SystemLayer` global with `DeviceLayer::SystemLayer()` indirecting through a pointer to make `System::Layer` actually mockable for code that uses PlatformManager. I left that out of this PR since it's a lot more search-and-replace churn in an already large PR.